Doc/library/collections.rst

```diff
@@ -1171,27 +1171,43 @@ original insertion position is changed and moved to the end::
             self.move_to_end(key)
 
 An :class:`OrderedDict` would also be useful for implementing
-variants of :func:`functools.lru_cache`::
+variants of :func:`functools.lru_cache`:
 
-    class LRU(OrderedDict):
-        'Limit size, evicting the least recently looked-up key when full'
+.. testcode::
 
-        def __init__(self, maxsize=128, /, *args, **kwds):
-            self.maxsize = maxsize
-            super().__init__(*args, **kwds)
+    class LRU:
 
-        def __getitem__(self, key):
-            value = super().__getitem__(key)
-            self.move_to_end(key)
+        def __init__(self, func, maxsize=128):
+            self.func = func
+            self.maxsize = maxsize
+            self.cache = OrderedDict()
+
+        def __call__(self, *args):
```

Comment:
I think it would be better to provide just two methods: `get()` and `set()`. It would be more simple and universal tool.

Reply:
The goal isn't to make a universal tool.  The goals are 1) showcase the *move_to_end()* and *popitem()* methods and 2) provide a basis for people to roll their own variants of *lru_cache()* using *OrderedDict()* to provide the least-recently-used algorithm and leaving other aspects of the API up to the developer.